### PR TITLE
Feat/middleware

### PR DESCRIPTION
**Pull request by delveper**

### CODE REVIEW

File: app/transport/http/middleware.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	}
+	return handler
+}
+
+// WithLogRequest will log detailed request info
+func WithLogRequest(logger *lgr.Logger) Middleware {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			defer logger.Infow("Request:",
+				"Header", req.Header,
```
Comment (vpakhuchyi) [inline app/transport/http/middleware.go:25]: So, what do we expect to receive in "Header"?

Comment (vpakhuchyi) [reply app/transport/http/middleware.go:25]: I mean, we print all headers now. Why?

Comment (vpakhuchyi) [reply app/transport/http/middleware.go:25]: Moreover, "User-Agent" is also comes from Header map, so its value is duplicated now.

Comment (vpakhuchyi) [reply app/transport/http/middleware.go:25]: Please recall the principal that we discussed on the previous call - minimalism.
You want to print only data that can be useful. 
Cos right now, if user add 100 custom headers - you will print them 😄

Comment (delveper) [reply app/transport/http/middleware.go]: Amateur talent activities were removed successfully :smiley:

**Review by vpakhuchyi — APPROVED** [review]

**Review by lifehou5e — APPROVED** [review]

**Review by Brigant — APPROVED** [review]

**Review by Rusli4k — APPROVED** [review]

**Review by Anton-Hudz — APPROVED** [review]

**Pull request merged**